aws_lambda_powertools/utilities/data_classes/s3_batch_operation_event.py

```diff
@@ -0,0 +1,237 @@
+import warnings
+from dataclasses import dataclass, field
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+from urllib.parse import unquote_plus
+
+from aws_lambda_powertools.shared.types import Literal
+from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
+
+# list of valid result code. Used both in S3BatchOperationResponse and S3BatchOperationResponseRecord
+VALID_RESULT_CODE_TYPES: Tuple[str, str, str] = ("Succeeded", "TemporaryFailure", "PermanentFailure")
+
+
+@dataclass(repr=False, order=False)
+class S3BatchOperationResponseRecord:
+    task_id: str
+    result_code: Literal["Succeeded", "TemporaryFailure", "PermanentFailure"]
```

Comment:
I see these strings repeated multiple times in this file, can we move them to constants so we can avoid typos?

Reply:
Moved.